common/authorization/oauthAuthorizer.go

```diff
@@ -101,49 +104,68 @@ func (a *oauthAuthority) Authorize(
 		return Result{Decision: DecisionDeny}, nil
 	}
 
-	claims, err := a.parseToken(token, a.verifier)
+	var claims JWTClaims
+
+	_, err := a.parser.ParseWithClaims(token, &claims, a.keyFunc)
+
 	if err != nil {
 		a.log.Debug("request is not authorized", tag.Error(err))
 		return Result{Decision: DecisionDeny}, nil
 	}
 
-	if err := a.validateTTL(claims); err != nil {
+	if err := a.validateTTL(&claims); err != nil {
 		a.log.Debug("request is not authorized", tag.Error(err))
 		return Result{Decision: DecisionDeny}, nil
 	}
 
 	if claims.Admin {
 		return Result{Decision: DecisionAllow}, nil
 	}
+
 	domain, err := a.domainCache.GetDomain(attributes.DomainName)
 	if err != nil {
 		return Result{Decision: DecisionDeny}, err
 	}
 
-	if err := validatePermission(claims, attributes, domain.GetInfo().Data); err != nil {
+	if err := validatePermission(&claims, attributes, domain.GetInfo().Data); err != nil {
 		a.log.Debug("request is not authorized", tag.Error(err))
 		return Result{Decision: DecisionDeny}, nil
 	}
 
 	return Result{Decision: DecisionAllow}, nil
 }
 
-func (a *oauthAuthority) parseToken(tokenStr string, verifier jwt.Verifier) (*JWTClaims, error) {
-	token, err := jwt.ParseAndVerifyString(tokenStr, verifier)
-	if err != nil {
-		return nil, fmt.Errorf("parse token: %w", err)
-	}
-	var claims JWTClaims
-	_ = json.Unmarshal(token.RawClaims(), &claims)
-	return &claims, nil
+// keyFunc returns correct key to check signature
+func (a *oauthAuthority) keyFunc(token *jwt.Token) (interface{}, error) {
+	// only local public key is supported currently
+	return a.publicKey, nil
 }
 
 func (a *oauthAuthority) validateTTL(claims *JWTClaims) error {
-	if claims.TTL > a.authorizationCfg.MaxJwtTTL {
-		return fmt.Errorf("token TTL: %d is larger than MaxTTL allowed: %d", claims.TTL, a.authorizationCfg.MaxJwtTTL)
+
+	if claims.IssuedAt == nil {
+		return errors.New("IssuedAt is not set")
+	}
+
+	// Fill ExpiresAt when TTL is passed
```

Comment:
I'm a little surprised this is necessary, it's not a problem, but I wouldj have throught that the JWT lib would do this natively

Reply:
Our clients (java and go) are not setting `ExpiresAt` claim. Instead, they are providing TTL claim. This is to support backwards compatibility. Later on, clients needs to be updated to fill in `ExpiresAt` and we can delegate this check to JWT lib.

Reply:
After re-reading your comment, I realized that it was not about TTL check, but for `IssuedAt` check. And yes, you are right, this is not needed as library will do this validation. Thanks!

Reply:
ah, sorry, yeah, i wasn't being super clear and to be honest, i get confused between the different values. I'll don't think it's a blocker, so stamping. Thank you for looking into it.